lib/block_view.go

```diff
@@ -2815,25 +2851,58 @@ func (bav *UtxoView) _disconnectNFTBid(
 	operationData := utxoOpsForTxn[operationIndex]
 	operationIndex--
 
+	// We sometimes have some extra AddUtxo operations we need to remove
+	// These are "implicit" outputs that always occur at the end of the
+	// list of UtxoOperations. The number of implicit outputs is equal to
+	// the total number of "Add" operations minus the explicit outputs.
+	numUtxoAdds := 0
+	for _, utxoOp := range utxoOpsForTxn {
+		if utxoOp.Type == OperationTypeAddUtxo {
+			numUtxoAdds += 1
+		}
+	}
+	operationIndex -= numUtxoAdds - len(currentTxn.TxOutputs)
+
 	// Get the NFTBidEntry corresponding to this txn.
 	bidderPKID := bav.GetPKIDForPublicKey(currentTxn.PublicKey)
 	if bidderPKID == nil || bidderPKID.isDeleted {
 		return fmt.Errorf("_disconnectNFTBid: PKID for bidder public key %v doesn't exist; this should never happen", string(currentTxn.PublicKey))
 	}
-	nftBidKey := MakeNFTBidKey(bidderPKID.PKID, txMeta.NFTPostHash, txMeta.SerialNumber)
-	nftBidEntry := bav.GetNFTBidEntryForNFTBidKey(&nftBidKey)
-	// Sanity-check that it exists.
-	if nftBidEntry == nil || nftBidEntry.isDeleted {
-		return fmt.Errorf("_disconnectNFTBid: Bid entry for "+
-			"nftBidKey %v doesn't exist; this should never happen", nftBidKey)
-	}
+	// If there isn't a previous NFT entry, we're dealing with a bid on a non-BuyNow NFT.
+	if operationData.PrevNFTEntry == nil {
```

Comment:
Nit: Feels like we should have a more robust/explicit way of knowing whether this bid was on a BuyNow NFT vs on a normal NFT. I could see us setting PrevNFTEntry for other reasons in the future, which would cause an unexpected bug to occur here if we don't make this if statement a little more narrow.

Reply:
Maybe you reverse the check and do:
```
if operationData.PrevNFTEntry != nil && operationData.PrevNFTEntry.IsBuyNow {
  // In this case, the bid was on a buy-now NFT, which means that we need to ...
} else {
  // In this case, we know that we're dealing with a non-buy-now NFT...
}
```